pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandlerTest.java

```diff
@@ -0,0 +1,432 @@
+      throws Exception {
+    List<FieldConfig> fieldConfigs = new ArrayList<>(_tableConfig.getFieldConfigList());
+    Random rand = new Random();
+    int randomIdx = rand.nextInt(_allCompressionTypes.size());
+    FieldConfig.CompressionCodec newCompressionType = _allCompressionTypes.get(randomIdx);
+
+    // Column 1
+    randomIdx = rand.nextInt(fieldConfigs.size());
+    FieldConfig config1 = fieldConfigs.remove(randomIdx);
+    String column1 = config1.getName();
+    FieldConfig newConfig1 =
+        new FieldConfig(column1, FieldConfig.EncodingType.RAW, Collections.emptyList(), newCompressionType, null);
+    fieldConfigs.add(newConfig1);
+
+    // Column 2
+    randomIdx = rand.nextInt(fieldConfigs.size());
+    FieldConfig config2 = fieldConfigs.remove(randomIdx);
+    String column2 = config2.getName();
+    FieldConfig newConfig2 =
+        new FieldConfig(column2, FieldConfig.EncodingType.RAW, Collections.emptyList(), newCompressionType, null);
+    fieldConfigs.add(newConfig2);
+
+    TableConfig tableConfig =
+        new TableConfigBuilder(TableType.OFFLINE).setTableName(TABLE_NAME).setNoDictionaryColumns(_noDictionaryColumns)
+            .setFieldConfigList(fieldConfigs).build();
+    tableConfig.setFieldConfigList(fieldConfigs);
+
+    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig(null, tableConfig);
+    IndexCreatorProvider indexCreatorProvider = IndexingOverrides.getIndexCreatorProvider();
+    ForwardIndexHandler fwdIndexHandler = new ForwardIndexHandler(_existingSegmentMetadata, indexLoadingConfig);
+    fwdIndexHandler.updateIndices(_writer, indexCreatorProvider);
+
+    // Validate column 1
+    ColumnMetadata columnMetadata = _existingSegmentMetadata.getColumnMetadataFor(column1);
+    validateForwardIndex(columnMetadata, newCompressionType);
+
+    // Validate column 2
+    columnMetadata = _existingSegmentMetadata.getColumnMetadataFor(column2);
+    validateForwardIndex(columnMetadata, newCompressionType);
+  }
+
+
+  private void validateForwardIndex(ColumnMetadata columnMetadata, FieldConfig.CompressionCodec expectedCompressionType)
+      throws IOException {
+    String columnName = columnMetadata.getColumnName();
+
+    // Check Compression type in header
+    ForwardIndexReader fwdIndexReader = LoaderUtils.getForwardIndexReader(_writer, columnMetadata);
+    ChunkCompressionType fwdIndexCompressionType = fwdIndexReader.getCompressionType();
+    assertEquals(fwdIndexCompressionType.name(), expectedCompressionType.name());
+
+    // Check index map if the forward index offsets are present.
+    String segmentDir = INDEX_DIR + "/" + SEGMENT_NAME + "/v3";
+    File idxMapFile = new File(segmentDir, V1Constants.INDEX_MAP_FILE_NAME);
+    FileUtils.readFileToString(idxMapFile, StandardCharsets.UTF_8).contains(columnName + ".forward_index.startOffset");
+    FileUtils.readFileToString(idxMapFile, StandardCharsets.UTF_8).contains(columnName + ".forward_index.size");
```

Comment:
Did you actually mean to assert that `contains`() should return true ? Looks like we are not doing anything about the return value at 409-410

Reply:
Done.